controllers/fenceagentsremediation_controller.go

```
@@ -79,7 +84,16 @@ func (r *FenceAgentsRemediationReconciler) Reconcile(ctx context.Context, req ct
 		r.Log.Error(err, "failed to get FAR CR")
 		return emptyResult, err
 	}
-	// TODO: Validate FAR CR name to nodeName. Run isNodeNameValid
+	// Validate FAR CR name to match a nodeName from the cluster
+	r.Log.Info("Check FAR CR's name")
+	if err := farUtils.CheckNodeName(r.Client, req.Name); err != nil {
+		if apiErrors.IsNotFound(err) {
+			return emptyResult, err
+		}
+		r.Log.Error(err, "Failed to validate FAR CR's name- retry")
+		return ctrl.Result{RequeueAfter: apiFailureRetryTimeout}, nil
```

Why not just return the error here ?